lib/rake/javaextensiontask.rb

```diff
@@ -212,13 +212,26 @@ def java_classpath_arg(*args)
         rescue => e
         end
       end
+
       unless jruby_cpath
         libdir = RbConfig::CONFIG['libdir']
         if libdir.start_with? "classpath:"
           raise 'Cannot build with jruby-complete'
         end
-        jruby_cpath = File.join(libdir, "jruby.jar")
+        candidate = File.join(libdir, "jruby.jar")
+        jruby_cpath = candidate if File.exist? candidate
       end
+
+      unless jruby_cpath
+        jruby_home = ENV['JRUBY_HOME']
+        if jruby_home
+          candidate = File.join(jruby_home, 'lib', 'jruby.jar')
+          jruby_cpath = candidate if File.exist? candidate
+        end
+      end
+
+      raise "jruby.jar path not found" unless jruby_cpath
```

Comment:
Can we show candidates we tried here? It'll be helpful for users who want to debug their environment.

Reply:
Changed the error message